### PR TITLE
add spliterator support to TreeNode implementations

### PR DESCRIPTION
**Pull request by pjfanning**
relates to https://github.com/FasterXML/jackson-core/pull/1369

* Probably needs the jackson-core PR merged first.
* I can add test coverage in a follow up PR or add it here if preferred.
* the initial state of the PR is for discussion about the source changes

### CODE REVIEW

File: src/main/java/tools/jackson/databind/JsonNode.java [hunk 1]
```diff
@@ -983,6 +994,18 @@ public Iterator<JsonNode> elements() {
...
 
+    /**
+     * Method for accessing all value nodes of this Node, iff
+     * this node is a JSON Array or Object node. In case of Object node,
+     * field names (keys) are not included, only values.
+     * For other types of nodes, returns empty <code>Spliterator</code>.
+     *
+     * @since 3.0
+     */
+    public Spliterator<JsonNode> elementsSpliterator() {
```
Comment (cowtowncoder) [inline src/main/java/tools/jackson/databind/JsonNode.java:1005]: Based on `valueStream()` method, should probably be `value(s?)Spliterator()`

File: src/main/java/tools/jackson/databind/JsonNode.java [hunk 2]
```diff
@@ -991,6 +1014,15 @@ public Iterator<Map.Entry<String, JsonNode>> fields() {
         return ClassUtil.emptyIterator();
     }
 
+    /**
+     * @return <code>Spliterator</code> that can be used to traverse all key/value pairs
+     *   for object nodes; empty spliterator (no contents) for other types
+     * @since 3.0
+     */
+    public Spliterator<Map.Entry<String, JsonNode>> fieldsSpliterator() {
```
Comment (cowtowncoder) [inline src/main/java/tools/jackson/databind/JsonNode.java:1022]: 3.0 uses term "property" everywhere, all references to "field" are changed.

But combination of property name and value, "entry" is used in `JsonNode`, see `entryStream()`.

So should be `entrySpliterator()`?

File: src/main/java/tools/jackson/databind/JsonNode.java (hunk 1 above)
Comment (JooHyukKim) [reply src/main/java/tools/jackson/databind/JsonNode.java:1005]: +1 on `valueSpliterator`.

File: src/main/java/tools/jackson/databind/JsonNode.java (hunk 2 above)
Comment (cowtowncoder) [reply src/main/java/tools/jackson/databind/JsonNode.java:1022]: Actually, let me think about this one: I may have chosen wrong term: instead "entry" should really use "property"/"properties, as per:

https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-6

will change `entryStream` to `propertyStream()`; and then we should have `propertySpliterator`.
Or, if we are decide to go with plurals, `propertiesSpliterator()`.

Comment (cowtowncoder) [reply src/main/java/tools/jackson/databind/JsonNode.java:1022]: Singular it is: will change to `propertySpliterator()`. Similar for value case.

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**